### ISSUE TITLE
Blank line for long label does not work at the start of a code block

### ISSUE THREAD

**Issue by fviolette** [expressive-code/expressive-code#248]
When using this feature: https://expressive-code.com/key-features/text-markers/#adding-long-labels-on-their-own-lines

If I leave a blank line at the start like the following:

````
```dart {"1. Ensure JavaScript is enabled": 1-2}

webViewController.setJavaScriptMode(JavaScriptMode.unrestricted);

ContentsquareWebViewWrapper(
  delegate: JSChannelWebViewWrapperDelegate(
    addJavaScriptChannel: (WebViewJSChannelHandler handler) {
      webViewController.addJavaScriptChannel(
        handler.channelName,
        onMessageReceived: (jsMessage) {
          handler.onMessageReceived(jsMessage.message);
        },
      );
    },
    builder: (context) {
      return WebViewWidget(controller: webViewController);
    },
  ),
);
```
````

I expect to get:

```dart
|   1. Ensure JavaScript is enabled
| webViewController.setJavaScriptMode(JavaScriptMode.unrestricted);
ContentsquareWebViewWrapper(
  delegate: JSChannelWebViewWrapperDelegate(

    addJavaScriptChannel: (WebViewJSChannelHandler handler) {
      webViewController.addJavaScriptChannel(
        handler.channelName,
        onMessageReceived: (jsMessage) {
          handler.onMessageReceived(jsMessage.message);
        },
      );
    },

    builder: (context) {
      return WebViewWidget(controller: webViewController);
    },
  ),
);
```

I get this instead:

<img width="764" alt="image" src="https://github.com/user-attachments/assets/07427f78-1ab1-4ec3-b6a0-c8e0ff62bb8f">

The code will jump a line as soon as I add a character on the first line.
Am I using the feature correctly, or could this be a bug?

**Comment by hippotastic** [comment]
Haha, oh my, I would have never expected this side-effect of the feature in ExpressiveCodeBlock that automatically trims leading and trailing lines consisting of only whitespace.

I have just checked various possible workarounds and there doesn't seem to be a good solution right now. You could work around this by putting a [zero-width space](https://unicode-explorer.com/c/200B) in the line you want to reserve for the label, which is one of the characters that does not get recognized as whitespace by JavaScript's `trim()` functions, but this character will also end up in the contents copied to the clipboard.

In the very near future, I will add a feature that allows adding annotations through code comments directly inside your code. You will then be able to write this instead to get the desired effect:

```js
// [!mark] 1. Ensure JavaScript is enabled
webViewController.setJavaScriptMode(JavaScriptMode.unrestricted);
```

Multiple lines can then also be marked by adding a relative target range specifier like `[!mark:3]` for 3 lines.

I hope it's ok for you to wait for this feature to drop. I'm actively working on the annotation comments feature since the last few months on a daily basis to get it right, and am very close to releasing it.

**Comment by fviolette** [comment]
Any non-visible workaround will do in the short-term (and zero-width space works OK in that regard).
As usual, thanks for the explanation and update!

**Comment by nerdymomocat** [comment]
> In the very near future, I will add a feature that allows adding annotations through code comments directly inside your code. You will then be able to write this instead to get the desired effect:

Sorry for jumping in! This is the most exciting news ever. Would `ins`, `del` and `collapse` work with this format too?

**Comment by hippotastic** [comment]
Yes, they all will! Sorry for the lack of updates recently, I was a bit distracted from my open source work by other duties. I'll finish this in January! :)

**Comment by nerdymomocat** [comment]
You are awesome. Thank you. I saw the branches and the commits and realized that they will all be covered and squealed 😂. Happy new year!

**Comment by nerdymomocat** [comment]
> Multiple lines can then also be marked by adding a relative target range specifier like [!mark:3] for 3 lines.

Quick note about this intended feature: Would it be more difficult to process a start-end delimiter rather than a line count based range, similar to
```
[!mark:start]
a = 5
b = 5
[!mark:end]
```
